File: backend/internal/api/dto/menu.go
package dto

// MenuCreateRequest 创建菜单
type MenuCreateRequest struct {
	AppKey    string                 `json:"app_key"`
	ParentID  *string                `json:"parent_id"`
	SpaceKey  string                 `json:"space_key"`
	Kind      string                 `json:"kind"`
	Path      string                 `json:"path" binding:"max=255"`
	Name      string                 `json:"name" binding:"required,max=100"`
	Component string                 `json:"component"`
	Title     string                 `json:"title" binding:"required,max=100"`
	Icon      string                 `json:"icon"`
	SortOrder int                    `json:"sort_order"`
	Meta      map[string]interface{} `json:"meta"`
	Hidden    bool                   `json:"hidden"`
}

// MenuUpdateRequest 更新菜单
type MenuUpdateRequest struct {
	AppKey    string                 `json:"app_key"`
	ParentID  *string                `json:"parent_id"` // 空字符串表示移至顶级；省略字段表示保持不变
	SpaceKey  string                 `json:"space_key"`
	Kind      string                 `json:"kind"`
	Path      string                 `json:"path" binding:"max=255"`
	Name      string                 `json:"name" binding:"max=100"`
	Component string                 `json:"component"`
	Title     string                 `json:"title" binding:"max=100"`
	Icon      string                 `json:"icon"`
	SortOrder int                    `json:"sort_order"`
	Meta      map[string]interface{} `json:"meta"`
	Hidden    bool                   `json:"hidden"`
}

File: frontend-fluentV2/src/shared/types/menu.ts
export type MenuNodeKind = 'directory' | 'entry' | 'external'

export interface MenuNodeMeta extends Record<string, unknown> {
  title?: string
  icon?: string
  link?: string
  activePath?: string
  accessMode?: string
  isHide?: boolean
  isIframe?: boolean
  isHideTab?: boolean
  keepAlive?: boolean
  fixedTab?: boolean
  isFullPage?: boolean
}

export interface MenuManageGroup {
  id: string
  name: string
  sortOrder: number
  status: string
}

export interface MenuNode {
  id: string
  parentId?: string
  manageGroupId?: string
  manageGroup?: MenuManageGroup
  spaceKey: string
  kind: MenuNodeKind
  path: string
  name: string
  title: string
  component: string
  icon: string
  sortOrder: number
  hidden: boolean
  meta: MenuNodeMeta
  children: MenuNode[]
}

export interface MenuPageBinding {
  pageKey: string
  name: string
  routePath: string
  routeName?: string
  component: string
  parentMenuId?: string
  accessMode?: string
  permissionKey?: string
  parentPageKey?: string
  pageType: string
  spaceKey?: string
  spaceKeys: string[]
}

export interface MenuNodeDetail extends MenuNode {
  parent?: Pick<MenuNode, 'id' | 'title' | 'path'>
  childCount: number
  linkedPages: MenuPageBinding[]
  permissionKeys: string[]
}

export interface MenuDeletePreview {
  mode: 'single' | 'cascade' | 'promote_children'
  menuCount: number
  childCount: number
  affectedPageCount: number
  affectedRelationCount: number
}

export interface MenuMutationDraft {
  parentId: string | null
  manageGroupId: string | null
  spaceKey: string
  kind: MenuNodeKind
  path: string
  name: string
  title: string
  component: string
  icon: string
  sortOrder: number
  hidden: boolean
  accessMode: string
  activePath: string
  externalLink: string
  keepAlive: boolean
  fixedTab: boolean
  isFullPage: boolean
}
